### ISSUE TITLE
$Keys will add some object's prototype chain function's name as key when that function be used. 

### ISSUE THREAD

**Issue by iamchenxin** [facebook/flow#2280]
as below [flow try](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVBjOA7AzgFzAFMBbMAXjAG9UwwBDRgLgHJH6WAaVAXwG5U+AJ4AHImACiAWQD6AaQkBNAMoUwAEjlEhuADzCxcKMRIA+AVjyEMARibT5S1ZRYA6ABb1cAeQTYACgBOcGKBwix8YMDAYPjuAJa4YAjxMDBgUPSptNHJ7kTYYAAmRFDx2OL0GQCu2Bj48TgMJDgA5rH5YFglqFC19Y2F+LgAFADWTGAEgeWtnMSTBkRGJgCUkwBGcHAwRPSFNHSBRPjVgYVEHl6+AcGhwuOrAjyoucpEpEmXnj5+QSFEMJCZKpdIkehjcRXX63AFAsAbcSWXDxErHIoMJJVSHAlakVyYHAELoAJnssgUKjUbh+N3+9yEESiMTiiRBaTAIi8uCAA) :

```
const em = {
  aaa:'aaa',
};
type EM_KEYS = $Keys<typeof em>;
const c1:EM_KEYS = '.hasOwnProperty'; // this will fail

 // when define a function among the code
function ts(k: string, e: typeof em): boolean {
  return e.hasOwnProperty(k);
}

// Seems e.hasOwnProperty will make .hasOwnProperty be considered as a key of em.
const c2:EM_KEYS = '.hasOwnProperty'; // this will pass
```

**Comment by vkurchatkin** [comment]
Here is a simplified test case:

```js
const a = { a: 'a' };

a.foo

const c2: $Keys<typeof a> = '.foo'; // notice the dot
```

**Comment by iamchenxin** [comment]
@vkurchatkin its right.
but im not sure if the a.foo will make Flow add a key is a designed behavior or not.
however Object's prototype chain functions may not be considered as a part of keys.

**Comment by vkurchatkin** [comment]
This seems to be a bug, but not a critical one. Because if a property 'foo' actually exists, '.foo' is not added

**Comment by iamchenxin** [comment]
This issue is fixed in Flow 0.33.
